Sources/Plasma/FeatureLib/pfGameGUIMgr/pfGUIMultiLineEditCtrl.cpp

```diff
@@ -1046,6 +1046,28 @@ void    pfGUIMultiLineEditCtrl::HandleMouseHover(hsPoint3& mousePt, uint8_t modi
     }
 }
 
+//// HandleMouseWheel /////////////////////////////////////////////////////////
+
+void pfGUIMultiLineEditCtrl::HandleMouseWheel(hsPoint3& mousePt, uint8_t modifiers)
+{
+    static float deltaAccum = 0.f;
+
+    float wheelAmount = mousePt.fZ;
+
+    // Set the Z-component to 0 for the bounds test
+    mousePt.fZ = 0;
+
+    if (IHandleMouse(mousePt)) {
+        deltaAccum += wheelAmount;
+        if (fabs(deltaAccum) >= 120.f) {
+            int32_t delta = -1 * int32_t(deltaAccum / 120.f);
+            SetScrollPosition(GetScrollPosition() + delta);
+
+            deltaAccum += (delta * 120.f);
+        }
+    }
+}
```

Comment:
It feels like the wheel delta accumulation should be handled before we get to the individual GUI controls. AFAIK, we don't really support partial scrolling anywhere, so it might be worth accumulating and sending out whole lines either in the input manager or the GUI manager... or perhaps having wheel accumulation being something in the base `pfGUIControlMod` class. Otherwise, we get duplicated accumulation code and interesting semantics where the different controls in the same type knowing about each others' wheel accumulation but different control types do not. That might make for some interesting behavior.

Reply:
Putting this in the GameGUIMgr might make sense.

Do you think the mousePt Z should contain the pre-divided delta (i.e., how many lines to scroll) or just the accumulated delta and let each control figure that out?

Reply:
That's a good question. I think both of those would be valid approaches, but I have a slight preference for sending the accumulated delta.